crates/flipperzero/src/furi/rng.rs

```diff
@@ -0,0 +1,49 @@
+use flipperzero_sys as sys;
+use rand_core::{impls, CryptoRng, Error, RngCore};
+
+/// A random number generator that retrieves randomness from the Flipper Zero hardware.
+///
+/// This is a zero-sized struct. It can be freely constructed with `HwRng`.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct HwRng;
+
+impl CryptoRng for HwRng {}
+
+impl RngCore for HwRng {
```

Comment:
Instead of (or in addition to) this approach, we could provide a standlone function that matches the API of `getrandom::getrandom`, and then the logic that sets up the root-level FAP could load that in as a [custom `getrandom` backend](https://docs.rs/getrandom/0.2.9/getrandom/macro.register_custom_getrandom.html). This would enable any usages of `getrandom::getrandom` to work, as well as `rand_core::OsRng` / `rand::rngs::OsRng`.